lib/ansible/plugins/action/__init__.py

```diff
@@ -1045,6 +1045,10 @@ def _low_level_execute_command(self, cmd, sudoable=True, in_data=None, executabl
             display.debug("_low_level_execute_command(): changing cwd to %s for this command" % chdir)
             cmd = self._connection._shell.append_command('cd %s' % chdir, cmd)
 
+        # https://github.com/ansible/ansible/issues/68054
+        if executable:
+            self._connection._shell.executable = executable
```

Comment:
never assign directly, use set_option(), in this case `direct={'executable'=executable} `

right now this is 'safe' due to using multi procs, we should add note about not being thread safe